src/main/java/org/piwik/java/tracking/PiwikTracker.java

```diff
@@ -103,17 +108,53 @@ public HttpResponse sendRequest(PiwikRequest request) throws IOException{
         }
     }
 
+    /**
+     * Send a request.
+     * @param request request to send
+     * @return future with response from this request
+     * @throws IOException thrown if there was a problem with this connection
+     */
+    public Future<HttpResponse> sendRequestAsync(PiwikRequest request) throws IOException{
+        CloseableHttpAsyncClient client = getHttpAsyncClient();
+        uriBuilder.setCustomQuery(request.getQueryString());
+        HttpGet get = null;
+
+        try {
+            get = new HttpGet(uriBuilder.build());
+            return client.execute(get,null);
+        } catch (URISyntaxException e) {
+            throw new IOException(e);
+        } finally {
+            if (get != null) {
```

Comment:
Is it ok that the connection will be released sooner than the actual request finishes?  Due to the async nature of the code, the execution will leave finish the try block as soon as the get Future is created, moving to the finally block, closing the connection. Meanwhile, somewhere in the background, the connection is still used by the client, I guess.